tensorboard/webapp/core/views/page_title_container.ts

```diff
@@ -36,12 +59,43 @@ import {State} from '../state';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class PageTitleContainer {
-  readonly title$ = this.store.pipe(
-    select(getEnvironment),
-    map((env) => env.window_title || 'TensorBoard'),
-    // (it's an empty string when the `--window_title` flag is not set)
+  private readonly getExperimentId$ = this.store.select(getRouteKind).pipe(
+    filter((routeKind) => routeKind === RouteKind.EXPERIMENT),
+    withLatestFrom(this.store.select(getExperimentIdsFromRoute)),
+    map(([, experimentIds]) =>
+      experimentIds && experimentIds.length === 1 ? experimentIds[0] : null
+    )
+  );
+
+  private readonly experimentName$ = this.getExperimentId$.pipe(
+    filter(Boolean),
+    mergeMap((experimentId) => {
+      return this.store.select(getExperiment, {experimentId});
+    }),
+    map((experiment) => (experiment ? experiment.name : null))
+  );
+
+  readonly title$ = this.store.select(getEnvironment).pipe(
+    combineLatestWith(this.experimentName$),
+    map(([env, experimentName]) => {
+      const tbBrandName = this.customBrandName
+        ? this.customBrandName
+        : 'TensorBoard';
+      if (env.window_title) {
+        // (it's an empty string when the `--window_title` flag is not set)
+        return env.window_title;
+      } else if (experimentName) {
+        return `${experimentName} - ${tbBrandName}`;
+      } else {
+        return tbBrandName;
+      }
+    }),
+    startWith(this.customBrandName ? this.customBrandName : 'TensorBoard'),
     distinctUntilChanged()
   );
 
-  constructor(private readonly store: Store<State>) {}
+  constructor(
+    private readonly store: Store<State>,
+    @Optional() @Inject(TB_BRAND_NAME) readonly customBrandName: string
```

Comment:
nit: is it possible to make `private readonly`?

Reply:
Done.